From: Yotam Gigi <yotamg@mellanox.com>
Date: Thu, 21 Jul 2016 12:03:16 +0200
Subject: net/sched: act_mirred: Add helper inlines to access tcf_mirred info.
Patch-mainline: v4.8-rc1
Git-commit: 56a20680f70393d199fc5e8ecc7859549ca5a0c0
References: bsc#1015342 FATE#321688 bsc#1015343 FATE#321689

The helper function is_tcf_mirred_mirror helps finding whether an action
struct is of type mirred and is configured to be of type mirror.

Signed-off-by: Yotam Gigi <yotamg@mellanox.com>
Signed-off-by: Jiri Pirko <jiri@mellanox.com>
Signed-off-by: David S. Miller <davem@davemloft.net>
Acked-by: Benjamin Poirier <bpoirier@suse.com>
---
 include/net/tc_act/tc_mirred.h |    9 +++++++++
 1 file changed, 9 insertions(+)

--- a/include/net/tc_act/tc_mirred.h
+++ b/include/net/tc_act/tc_mirred.h
@@ -24,6 +24,15 @@ static inline bool is_tcf_mirred_redirec
 	return false;
 }
 
+static inline bool is_tcf_mirred_mirror(const struct tc_action *a)
+{
+#ifdef CONFIG_NET_CLS_ACT
+	if (a->ops && a->ops->type == TCA_ACT_MIRRED)
+		return to_mirred(a)->tcfm_eaction == TCA_EGRESS_MIRROR;
+#endif
+	return false;
+}
+
 static inline int tcf_mirred_ifindex(const struct tc_action *a)
 {
 	return to_mirred(a)->tcfm_ifindex;
